tools/wpt/browser.py

```diff
@@ -179,22 +180,70 @@ def find_certutil(self):
     def find_webdriver(self):
         return find_executable("geckodriver")
 
-    def install_prefs(self, dest=None):
+    def get_version_number(self, binary):
+        version_re = re.compile("Mozilla Firefox (\d+\.\d+(?:\.\d+)?)(a|b)?")
+        proc = subprocess.Popen([binary, "--version"], stdout=subprocess.PIPE)
+        stdout, _ = proc.communicate()
+        stdout.strip()
+        m = version_re.match(stdout)
+        if not m:
+            return None, "nightly"
+        version, status = m.groups()
+        channel = {"a": "nightly", "b": "beta", None: "stable"}
+        return version, channel.get(status, "stable")
```

Comment:
I don't know which of the two approaches to defining the default value is more Pythonic, but I don't think we need them both.